message_definitions/v1.0/messages.xml

```diff
@@ -96,6 +96,10 @@
       <field name="pos_y" type="float" unit="m" format="%.1f"/>
       <field name="dist_wp" type="float" format="%.1f" unit="m"/>
       <field name="dist_home" type="float" format="%.1f" unit="m"/>
+      <field name="flight_time" type="uint16" unit="s"/>
```

Comment:
Adding block and stage time here is nice. 

It might be better to add the time to the firmware status messages instead of the nav/flight plan messages given that it is not strictly coupled to the nav/flight plan. It is already in the AP for fixedwing, you can change the rotorcraft one from cpu_time to flight time as that seems more useful. 

I am not sure what the kill_auto_throttle is used for, can't find it with a quick search so I guess it's fine.

Reply:
flight time is usually started by and in the flight plan for fixedwing (and it doesn't stop after landing actually), so it is a bit different from rotorcraft where "in_flight" state is relatively easy to guess.

the kill_throttle is the boolean that make the throttle indicator turn red in GCS strip, meaning that the navigation is not able to use throttle anymore. This one is really important for safety.

In general I don't like the AP / NAV status message, but changing that is a bigger issue, we need to fix this quickly for now.

Reply:
this is not critical so I'm fine with it. Maybe make an issue so someone does this in the future.